codegen/module_system.go

```diff
@@ -65,44 +63,46 @@ type StructMeta struct {
 type EndpointTestMeta struct {
 	Instance           *ModuleInstance
 	Method             *MethodSpec
-	TestStubs          []TestStub
+	TestFixtures       map[string]*EndpointTestFixture `json:"testFixtures"`
 	ClientName         string
 	ClientID           string
 	RelativePathToRoot string
 	IncludedPackages   []GoPackageImport
 }
 
-// TestStub saves stubbed requests/responses for an endpoint test.
-type TestStub struct {
-	TestName               string
-	EndpointID             string
-	HandlerID              string
-	EndpointRequest        map[string]interface{} // Json blob
-	EndpointRequestString  string
-	EndpointReqHeaders     map[string]string      // Json blob
-	EndpointReqHeaderKeys  []string               // To keep in canonical order
-	EndpointResponse       map[string]interface{} // Json blob
-	EndpointResponseString string
-	EndpointResHeaders     map[string]string // Json blob
-	EndpointResHeaderKeys  []string          // To keep in canonical order
-
-	ClientStubs []ClientStub
-
-	TestServiceName string // The service module that mounts the endpoint
+// FixtureBlob is map[string]interface{} that implements default string
+// used for headers and (http | tchannel) request/response
+type FixtureBlob map[string]interface{}
+
+// String convert map[string]interface{} to string
+func (fb *FixtureBlob) String() string {
+	if str, err := json.Marshal(fb); err == nil {
+		return string(str)
+	}
+	return ""
```

Comment:
the error for marshaling is dropped here silently. let's at least return `err.Error()`

Reply:
nope this is used in template, template is dumb don't know how to handle it, actually this should be `{}`